qubes/utils.py

```diff
@@ -246,3 +246,30 @@ async def void_coros_maybe(values):
         done, _ = await asyncio.wait(coros)
         for task in done:
             task.result()  # re-raises exception if task failed
+
+# pylint: disable=redefined-builtin
+async def run_program(*args, check=False, input=None, **kwargs):
+    """Async version of subprocess.run()
+    """
+    p = await asyncio.create_subprocess_exec(*args, **kwargs)
+    stdouterr = await p.communicate(input=input)
+    if check and p.returncode:
+        raise subprocess.CalledProcessError(p.returncode,
+                                                args[0], *stdouterr)
+    return p
+
+def cryptsetup(*args):
+    """
+    Run cryptsetup with the given arguments.  This method returns a future.
+    """
+    return run_program(
+        'sudo',
+        'cryptsetup',
+        *args,
+        executable='/usr/bin/sudo',
```

Comment:
Please spare the logs and use sudo only when necessary - as you did in #399. Actually, it makes sense to use the same mechanism (and I consider #399 a prerequisite, because otherwise this PR will outright fail to start a VM in a file pool). So, lets wait for merging #399, and then fix this while doing rebase.

Reply:
Why will this PR otherwise fail to start a VM in a file pool?  Volatile volumes do not use a custom block script even with the current file pool, IIRC.  Non-volatile volumes do, but this PR does not alter how those are handled.

Reply:
Ah, indeed volatile volumes don't use script. Ok, lets see what openQA says (it doesn't supersedes the need for unit tests).